libnestutil/type_traits.h

```diff
@@ -535,7 +535,7 @@ struct is_same< T, T > : public true_type
 };
 
 // Specified by TR1 [4.6] Relationships between types
-#if !defined( _MSC_VER ) && !( defined( __GNUC__ ) && __GNUC__ <= 3 )
+#if not defined( _MSC_VER ) && not( defined( __GNUC__ ) && __GNUC__ <= 3 )
```

Comment:
'not' isn't supported by the preprocessor language. Since 'not' is syntactically correct (it is a valid macro name), the preprocessor will generate a warning in the best case. 'not' should be replaced with '!' or the expression can also be re-written using #ifndef.